src/Sarif/Schemata/sarif-schema.json

```diff
@@ -303,11 +303,6 @@
           "$ref": "#/definitions/externalPropertyFile"
         },
 
-        "resources": {
```

Comment:
>resources [](start = 9, length = 9)

It's right to remove this -- but now I think you need to allow "tool" to be externalized. Either that, or externalize `tool.rulesMetadata`, `tools.notificationsMetadata`, and `tools.globalMessageStrings` separately -- then we would have to bite the bullet and specify the JPath for each of them (up to now, every externalized property has come directly from the `run` object).

Reply:
ok, good catch. for now, I've externalized run.tool. this isn't a coherent solution because our schema requires some representation of tool to be a valid log file (presumably we could specify that the external property merges properly with it, dunno). anyway, i'm not worrying too much about this now, as we have a plan to hoist this thing to the root and drive it against JSON paths, as you noted.

---
In reply to: [255308008](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255308008) [](ancestors = 255308008)